src/js/events.js

```diff
@@ -268,163 +255,159 @@ jQuery( document ).ready( function() {
           jQuery( this ).dialog( 'close' );
         }
       }
-    });
-
+    } );
   } );
 
   // calculate and list rational approximations within user parameters
   jQuery( "#input_interval_to_approximate" ).change( function() {
     var interval = line_to_decimal( jQuery ( "#input_interval_to_approximate" ).val() );
-    
-        current_approximations.convergent_indicies = [];
-        current_approximations.numerators = [];
-        current_approximations.denominators = [];
-        current_approximations.ratios = [];
-        current_approximations.numerator_limits = [];
-        current_approximations.denominator_limits = [];
-        current_approximations.ratio_limits = [];
-
-        get_rational_approximations(interval, current_approximations.numerators, current_approximations.denominators, 999999,
-                                    current_approximations.convergent_indicies,
-                                    current_approximations.ratios,
-                                    current_approximations.numerator_limits,
-                                    current_approximations.denominator_limits,
-                                    current_approximations.ratio_limits);
-        modify_update_approximations();
-  });
-  
-  // recalculate approximations when scale degree changes
-  jQuery( "#input_scale_degree").change( function() {
-  jQuery( "#txt_tuning_data" ).val( jQuery( "#txt_tuning_data" ).val().trim() );
+    current_approximations.convergent_indicies = [];
+    current_approximations.numerators = [];
+    current_approximations.denominators = [];
+    current_approximations.ratios = [];
+    current_approximations.numerator_limits = [];
+    current_approximations.denominator_limits = [];
+    current_approximations.ratio_limits = [];
+
+    get_rational_approximations(
```

Comment:
If one of the main source of parameters for the `get_rational_approximations` is the `current_approximations` object, then the function should be refactored in a way, that it accepts the whole object and not individual values.

Reply:
I'd like to keep get_rational_approximations the same because sometimes I just want numerators and denominators, so I wouldn't want to pass in the whole object.

I added another helper function that lets you pass in an interval, and it automatically puts it in the current_approximations object. Pretty much moving that block of code from "events" to "sequences"